packages/common/src/ContractInteractor.ts

```diff
@@ -476,6 +477,7 @@ export class ContractInteractor {
    * In case 'getLogs' returned with a common error message of "more than X events" dynamically decrease page size.
    */
   async _getPastEventsPaginated (contract: any, names: EventName[], extraTopics: string[], options: PastEventOptions): Promise<EventData[]> {
+    const delay = this.chain === 'private' ? 0 : 300
```

Comment:
Can you explain the idea of adding a delay here?
Also, I do not like declaring a variable on the first line of a method only to use it on the last line. Can you move the declaration to where this is used?